### PR TITLE
Turns miming into a trait instead of an unsightly mind boolean

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

https://github.com/tgstation/tgstation/pull/72254 turned to be wildly unpopular so i'm simplifying it and keeping only the important, code backend stuff

## Why It's Good For The Game

makes the code gooder, makes mobs without a mind capable of miming (if that becomes necessary for whatever reason)

## Changelog

Not player facing

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
test failures are not my fucking fault

File: code/modules/spells/spell_types/self/mime_vow.dm [hunk 1]
```diff
@@ -10,16 +10,26 @@
 	school = SCHOOL_MIME
 	cooldown_time = 5 MINUTES
 
-	spell_requirements = SPELL_REQUIRES_HUMAN|SPELL_REQUIRES_MIND
+	spell_requirements = SPELL_REQUIRES_HUMAN
 	spell_max_level = 1
 
+/datum/action/cooldown/spell/vow_of_silence/Grant(mob/grant_to)
+	. = ..()
+	ADD_TRAIT(grant_to, TRAIT_MIMING, "[type]")
```
[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/modules/spells/spell_types/self/mime_vow.dm:18]: I'll deal with that soon.

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -470,6 +470,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 // achievement. Can also be used on rod-form wizards.
 // Normally only present in the mind of a Research Director.
 #define TRAIT_ROD_SUPLEX "rod_suplex"
+/// The mob has an active mime vow of silence, and thus is unable to speak and has other mime things going on
+#define TRAIT_MIMING "miming"
```
[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/__DEFINES/traits.dm:474]: oops i forgot

**Comment by ChungusGamer666** [comment]
i'll get to it soonish bitch

File: code/modules/spells/spell_types/self/mime_vow.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/modules/spells/spell_types/self/mime_vow.dm:18]: Actually no, it checks for the trait alone, this doesn't require mobs to have the spell at all.

File: code/_globalvars/traits.dm [hunk 3]
```diff
@@ -179,6 +179,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_CANNOT_BE_UNBUCKLED" = TRAIT_CANNOT_BE_UNBUCKLED,
 		"TRAIT_GAMER" = TRAIT_GAMER,
 		"TRAIT_UNKNOWN" = TRAIT_UNKNOWN,
+		"TRAIT_MIMING" = TRAIT_UNKNOWN,
```
Comment (jlsnow301) [inline code/_globalvars/traits.dm:182]: should this be TRAIT_MIMING?

Comment (ChungusGamer666) [reply code/_globalvars/traits.dm:182]: oh fuck yes it should

File: code/modules/mob/living/emote.dm [hunk 4]
```diff
@@ -263,7 +263,7 @@
 /datum/emote/living/laugh/get_sound(mob/living/user)
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(H.dna.species.id == SPECIES_HUMAN && (!H.mind || !H.mind.miming))
+		if(H.dna.species.id == SPECIES_HUMAN && !HAS_TRAIT(H, TRAIT_MIMING))
 			if(user.gender == FEMALE)
 				return 'sound/voice/human/womanlaugh.ogg'
 			else
```
Comment (jlsnow301) [inline code/modules/mob/living/emote.dm:269]:
```
	if(!ishuman(user))
		return
	var/mob/living/carbon/human/joker = user
	if(joker.dna.species.id != SPECIES_HUMAN || HAS_TRAIT(joker, TRAIT_MIMING))
		return
	if(user.gender == FEMALE)
		return 'sound/voice/human/womanlaugh.ogg'
		
	return pick('sound/voice/human/manlaugh1.ogg', 'sound/voice/human/manlaugh2.ogg')
```

**Review by jlsnow301 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/emote.dm (hunk 4 above)
Comment (ChungusGamer666) [reply code/modules/mob/living/emote.dm:269]: ok this is a nitpick but sure battlebuddy

Comment (jlsnow301) [reply code/modules/mob/living/emote.dm:269]: agreed, so I wrote it for ya pardner

**Pull request merged**